src/Functions/FunctionsStringDistance.cpp

```diff
@@ -226,6 +226,34 @@ struct ByteJaccardIndexImpl
 
 static constexpr size_t max_string_size = 1u << 16;
 
+void parseUTF8String(PaddedPODArray<UInt32> & code_points, const char * __restrict data, size_t size)
+{
+    const char * end = data + size;
+    while (data < end)
+    {
+        size_t len = UTF8::seqLength(*data);
+        if (len == 1)
+        {
+            code_points.push_back(static_cast<UInt32>(*data));
+            ++data;
+        }
+        else
+        {
+            auto code_point = UTF8::convertUTF8ToCodePoint(data, end - data);
+            if (code_point.has_value())
+            {
+                code_points.push_back(code_point.value());
+                data += len;
+            }
+            else
+            {
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "Illegal UTF-8 sequence, while processing '{}'", StringRef(data, end - data));
+            }
+        }
+    }
+}
```

Comment:
Looks very similar to parsing a string in `ByteJaccardIndexImpl`, do you think we can deduplicate it a little bit?

Reply:
ok, let me try